### PR TITLE
Update featureset spec for featureset source

### PR DESCRIPTION
**Pull request by anyalee0221**
# Description

Update featureset spec schema to support featureset feature source.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by fredms — APPROVED** [review]

**Review by runhli — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/test_configs/feature_set/invalid_spec4/FeatureSetSpec.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+source:
```
Comment (qjxu) [inline sdk/ml/azure-ai-ml/tests/test_configs/feature_set/invalid_spec4/FeatureSetSpec.yaml:1]: can you leave a comment what is invalid for this example?

File: sdk/ml/azure-ai-ml/tests/test_configs/feature_set/invalid_spec5/FeatureSetSpec.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+source:
```
Comment (qjxu) [inline sdk/ml/azure-ai-ml/tests/test_configs/feature_set/invalid_spec5/FeatureSetSpec.yaml:1]: can you leave a comment what is invalid for this example?

File: sdk/ml/azure-ai-ml/tests/test_configs/feature_set/invalid_spec3/FeatureSetSpec.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+source:
```
Comment (qjxu) [inline sdk/ml/azure-ai-ml/tests/test_configs/feature_set/invalid_spec3/FeatureSetSpec.yaml:1]: can you leave a comment what is invalid for this example?

**Review by qjxu — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/test_configs/feature_set/invalid_spec4/FeatureSetSpec.yaml (hunk 1 above)
Comment (anyalee0221) [reply sdk/ml/azure-ai-ml/tests/test_configs/feature_set/invalid_spec4/FeatureSetSpec.yaml:2]: Added.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/featureset_spec_metadata.py [hunk 4]
```diff
@@ -74,6 +75,16 @@ def _load(
         res: FeaturesetSpecMetadata = load_from_dict(
             FeaturesetSpecMetadataSchema, yaml_data, context, "", unknown=INCLUDE, **kwargs
         )
+        if res.source.type != "featureset":
```
[the remark this answers is not in the document]
Comment (anyalee0221) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/featureset_spec_metadata.py]: Moved.

**Pull request merged**